### PR TITLE
Feat: Login and Signup page

### PR DESCRIPTION
**Pull request by yashpatel1797**
Pages add
1. Login
2. Signup

[Live link](https://joyful-rolypoly-31b73d.netlify.app/)

Login page
image

Signup page
image

### CODE REVIEW

File: src/pages/Login/Login.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    const navigate = useNavigate()
+    const { authDispatch } = useAuth();
+
+    const [{ email, password }, loginDispatch] = useReducer(loginFormReducer, { email: "", password: "" })
+
+
+    const [togglePassword, setTogglePassword] = useState(false);
+    const showPassword = togglePassword ? "visibility_off" : "visibility";
+    const passwordVisibilityHandler = () => {
+        setTogglePassword(!togglePassword)
```
Comment (ravikumar0403) [inline src/pages/Login/Login.js]: If you state update is dependent on previous value then use the functional way of updating state

File: src/pages/Login/Login.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+            const response = await axios.post("api/auth/login", { email, password });
+
+
+            localStorage.setItem("token", response.data.encodedToken)
+            localStorage.setItem('userData', JSON.stringify(response.data.foundUser));
+            authDispatch({ type: "USER_LOGIN" })
+            authDispatch({ type: "TOKEN_ADD", payload: response.data.encodedToken })
+            authDispatch({ type: "USER_DATA_ADD", payload: response.data.foundUser })
+        } catch (error) {
```
Comment (ravikumar0403) [inline src/pages/Login/Login.js:39]: You can use single dispatch to perform all these three changes by passing appropriate payload

File: src/pages/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            console.log(error);
+        } finally {
+            navigate("/")
```
Comment (ravikumar0403) [inline src/pages/Login/Login.js]: If you use finally, it will run even if there is an error. That would not be the right ux

**Review by ravikumar0403 — COMMENTED** [review]
I've suggested a few changes which can be improved

**Review by shashikantmaurya0007 — APPROVED** [review]
Ravi has already done the detailed review.rest looks clean and beautiful

File: src/pages/Login/Login.js (hunk 3 above)
Comment (yashpatel1797) [reply src/pages/Login/Login.js]: yes, haven't done error handling yet will update this.

File: src/pages/Login/Login.js (hunk 2 above)
Comment (yashpatel1797) [reply src/pages/Login/Login.js:39]: Thanks for the suggestion. will implement this later.

**Pull request merged**